src/Microsoft.ApplicationInsights.AspNet/Config/TelemetryConfigurationSource.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.Config
+{
+    using Microsoft.Framework.ConfigurationModel;
+    using System;
+    using System.Collections.Generic;
+
+    internal class TelemetryConfigurationSource : IConfigurationSource
+    {
+        /// <summary>
+        /// Developer mode setting value.
+        /// </summary>
+        public bool? DeveloperMode { get; set; }
+
+        /// <summary>
+        /// Instrumentation key setting value.
+        /// </summary>
+        public string InstrumentationKey { get; set; }
+
+        /// <summary>
+        /// Endpoint address setting value.
+        /// </summary>
+        public string EndpointAddress { get; set; }
+
+        public void Load()
+        {
+        }
+
+        public IEnumerable<string> ProduceSubKeys(IEnumerable<string> earlierKeys, string prefix, string delimiter)
+        {
+            return null;
+        }
+
+        public void Set(string key, string value)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool TryGet(string key, out string value)
```

Comment:
This logic appears to be duplicated in the `TelemetryConfigurationSource` and `AddTelemetryConfiguration`. Why do we need this class? Can we simplify this by having the `AddApplicationInsightsSettings` use `IConfigurationSourceRoot.Set` instead of this class?